File: simplified-app-shared/src/main/java/org/nypl/simplified/app/reader/ReaderColorSchemes.java
package org.nypl.simplified.app.reader;

import android.graphics.Color;

import com.io7m.jnull.NullCheck;
import com.io7m.junreachable.UnreachableCodeException;

import org.nypl.simplified.reader.api.ReaderColorScheme;

/**
 * Color schemes.
 */

public final class ReaderColorSchemes {

  private ReaderColorSchemes()
  {

  }

  public static int background(final ReaderColorScheme scheme)
  {
    NullCheck.notNull(scheme, "Scheme");

    switch (scheme) {
      case SCHEME_BLACK_ON_BEIGE:
        return Color.argb(0xff, 242, 228, 203);
      case SCHEME_BLACK_ON_WHITE:
        return 0xff000000 | (Color.WHITE & 0xffffff);
      case SCHEME_WHITE_ON_BLACK:
        return 0xff000000 | (Color.BLACK & 0xffffff);
    }

    throw new UnreachableCodeException();
  }

  public static int foreground(final ReaderColorScheme scheme)
  {
    NullCheck.notNull(scheme, "Scheme");

    switch (scheme) {
      case SCHEME_BLACK_ON_BEIGE:
        return 0xff000000 | (Color.BLACK & 0xffffff);
      case SCHEME_BLACK_ON_WHITE:
        return 0xff000000 | (Color.BLACK & 0xffffff);
      case SCHEME_WHITE_ON_BLACK:
        return 0xff000000 | (Color.WHITE & 0xffffff);
    }

    throw new UnreachableCodeException();
  }
}

File: simplified-app-shared/src/main/java/org/nypl/simplified/app/reader/ReaderColorSchemes.kt
package org.nypl.simplified.app.reader

import android.graphics.Color
import androidx.annotation.ColorInt
import org.nypl.simplified.reader.api.ReaderColorScheme

/**
 * Color schemes.
 */

object ReaderColorSchemes {

  /**
   * Retrieve the background color for the given scheme as an Android ARGB color int.
   */

  @JvmStatic
  @ColorInt
  fun backgroundAsAndroidColor(scheme: ReaderColorScheme): Int =
    when (scheme) {
      ReaderColorScheme.SCHEME_BLACK_ON_BEIGE ->
        Color.argb(0xff, 0xf2, 0xe4, 0xcb)
      ReaderColorScheme.SCHEME_BLACK_ON_WHITE ->
        Color.argb(0xff, 0xff, 0xff, 0xff)
      ReaderColorScheme.SCHEME_WHITE_ON_BLACK ->
        Color.argb(0xff, 0x00, 0x00, 0x00)
    }

  /**
   * Retrieve the foreground color for the given scheme as an Android ARGB color int.
   */

  @JvmStatic
  @ColorInt
  fun foregroundAsAndroidColor(scheme: ReaderColorScheme): Int =
    when (scheme) {
      ReaderColorScheme.SCHEME_BLACK_ON_BEIGE ->
        Color.argb(0xff, 0x00, 0x00, 0x00)
      ReaderColorScheme.SCHEME_BLACK_ON_WHITE ->
        Color.argb(0xff, 0x00, 0x00, 0x00)
      ReaderColorScheme.SCHEME_WHITE_ON_BLACK ->
        Color.argb(0xff, 0xff, 0xff, 0xff)
    }

  /**
   * Retrieve the foreground color for the given scheme as a web '#rrggbb' string.
   */

  @JvmStatic
  fun backgroundAsBrowserHex(scheme: ReaderColorScheme): String =
    when (scheme) {
      ReaderColorScheme.SCHEME_BLACK_ON_BEIGE -> "#f2e4cb"
      ReaderColorScheme.SCHEME_BLACK_ON_WHITE -> "#ffffff"
      ReaderColorScheme.SCHEME_WHITE_ON_BLACK -> "#000000"
    }

  /**
   * Retrieve the foreground color for the given scheme as a web '#rrggbb' string.
   */

  @JvmStatic
  fun foregroundAsBrowserHex(scheme: ReaderColorScheme): String =
    when (scheme) {
      ReaderColorScheme.SCHEME_BLACK_ON_BEIGE -> "#000000"
      ReaderColorScheme.SCHEME_BLACK_ON_WHITE -> "#000000"
      ReaderColorScheme.SCHEME_WHITE_ON_BLACK -> "#ffffff"
    }
}
